src/api/utils/payload.ts

```diff
@@ -24,52 +26,35 @@ export function containsBinaryData(data: any): boolean {
     }
 
     return false;
-}
+};
 
-export function objectToFormData(obj: any, rootName = "", ignoreList: string[] = null, keyFormatter = decamelize) {
+export const objectToFormData = (obj: unknown, keyFormatter = snakeCase) => {
     const formData = new FormData();
 
-    function isBlob(data: any): boolean {
-        if (isObject(data)) {
-            return typeof data.size === "number" && typeof data.type === "string" && typeof data.slice === "function";
-        }
-        return false;
-    }
-
-    function ignore(root) {
-        return (
-            Array.isArray(ignoreList) &&
-            ignoreList.some(function (x) {
-                return x === root;
-            })
-        );
-    }
-
-    function appendFormData(data: any, root = "") {
-        if (!ignore(root)) {
-            if (isBlob(data)) {
-                formData.append(keyFormatter(root), data);
-            } else if (Array.isArray(data)) {
-                for (let i = 0; i < data.length; i++) {
-                    appendFormData(data[i], `${root}[${i}]`);
-                }
-            } else if (isObject(data) && data && !Buffer.isBuffer(data)) {
-                for (const key in data) {
-                    if (root === "") {
-                        appendFormData(data[key], key);
-                    } else {
-                        appendFormData(data[key], `${root}.${key}`);
-                    }
-                }
-            } else {
-                if (data !== null && typeof data !== "undefined") {
-                    formData.append(keyFormatter(root), data);
-                }
+    const appendFormData = (data: unknown, root = "", key = "") => {
+        const formattedKey = keyFormatter(key);
+        const path = !root ? formattedKey : formattedKey ? `${root}.${formattedKey}` : root;
+
+        if (isBlob(data)) {
+            // Blob
+            formData.append(path, data);
+        } else if (Array.isArray(data)) {
+            // Array
```

Comment:
unnecessary comments? 🤔

Reply:
Hum... I jut tried to list the cases as it was not obvious that the last one was for string and buffers. I can remove the comments if you think it is extra though 🙂

Reply:
doesn't hurt to keep the comments :meow_party: